apps/passwd.c

```diff
@@ -47,6 +33,14 @@ static unsigned const char cov_2char[64] = {
     0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A
 };
 
+static const char ascii_dollar[] = { 0x24, 0x00 };
+#if defined(CHARSET_EBCDIC) && !defined(CHARSET_EBCDIC_TEST)
+static const char ebcdic_dollar[] = { 0x5B, 0x00 };
+# define dollar ebcdic_dollar
+#else
+# define dollar ascii_dollar
+#endif
```

Comment:
That provides a way to configure with `-DCHARSET_EBCDIC` as a test on non-EBCDIC systems, which is what I did to verify that the code did the job correctly (with quite a bit of `iconv` on the command line, nothing I'd want to reproduce in a test recipe=